src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilation.cs

```diff
@@ -25,9 +26,10 @@ namespace ILCompiler
 {
     public sealed class LLVMCodegenCompilation : RyuJitCompilation
     {
+        private ConcurrentDictionary<int, CorInfoImpl> _corinfos;
         private readonly Dictionary<TypeDesc, LLVMTypeRef> _llvmStructs = new();
-        private readonly ConditionalWeakTable<Thread, CorInfoImpl> _corinfos = new();
         private string _outputFile;
```

Comment:
CWT is not suitable for us because we need to call the Jit after all of the threads have finished work. With CWT, some of the threads might have exited already and their `CorInfoImpl`s - collected.